### PR TITLE
NOISSUE - Add local alarms for events

### PR DESCRIPTION
**Pull request by anovakovic01**
Add alarm notifications for events that are currently active or about to start.

### CODE REVIEW

File: app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java [hunk 1]
```diff
@@ -60,7 +68,21 @@ void init() {
     @IgnoreWhen(IgnoreWhen.State.VIEW_DESTROYED)
     @Override
     public void accept(List<Event> events) throws Exception {
+        final String action = getString(R.string.alarm_id);
         eventsAdapter.setEvents(events);
+        for (Event event : events) {
+            if (event.getEndDate().getTime() > System.currentTimeMillis()) {
```
Comment (alexkahriman) [inline app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java]: Consider inverting if statement.

File: app/src/main/java/com/ftn/trippleaproject/system/AlarmBroadcastReceiver.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        final Notification notification = new NotificationCompat.Builder(context, channelId)
+                .setAutoCancel(true)
+                .setSmallIcon(R.drawable.ic_event)
+                .setContentTitle(title)
+                .setContentText(description)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setContentIntent(PendingIntent.getActivity(context, 0, intent, 0)) // necessary evil
+                .build();
+
+        final int id = (sid == null) ? 0 : Integer.valueOf(sid);
```
Comment (alexkahriman) [inline app/src/main/java/com/ftn/trippleaproject/system/AlarmBroadcastReceiver.java]: sid?

Comment (anovakovic01) [reply app/src/main/java/com/ftn/trippleaproject/system/AlarmBroadcastReceiver.java]: Yes, `sid`. (String ID)

File: app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java (hunk 1 above)
Comment (anovakovic01) [reply app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java]: What do you mean by inverting `if` statement? Is there logical error?

File: app/src/main/java/com/ftn/trippleaproject/system/AlarmBroadcastReceiver.java (hunk 2 above)
Comment (alexkahriman) [reply app/src/main/java/com/ftn/trippleaproject/system/AlarmBroadcastReceiver.java]: Sid sounds like an incurable disease, but ok, why not :)

File: app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java (hunk 1 above)
Comment (alexkahriman) [reply app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java]: No, it's a matter of indentation, if you invert the logic and say **continue** you can lessen indentation.

Comment (anovakovic01) [reply app/src/main/java/com/ftn/trippleaproject/ui/fragment/EventsFragment.java]: Although I agree with your observation that is not Java style, but Go and Swift style.

**Pull request merged**